packages/language-service/src/lib/utils/wrap-and-log.ts

```diff
@@ -0,0 +1,34 @@
+type FunctionConfig<T> = Pick<
+    {
+        [K in keyof T]: T[K] extends (...args: any[]) => infer R ? R : never;
+    },
+    {
+        [K in keyof T]: T[K] extends (...args: any[]) => any ? K : never;
+    }[keyof T]
+>;
+
+export function wrapAndCatchErrors<T extends Record<string, any>>(
+    config: Partial<FunctionConfig<T>>,
+    context: new (...args: any[]) => T
+) {
+    const proto = context.prototype;
+    for (const [name, defaultReturn] of Object.entries(config)) {
+        const func = proto[name];
+        if (typeof func !== 'function') {
+            console.error(
+                `expected to find a function named ${name} on context, but found ${typeof func}`
+            );
+            continue;
+        }
+        proto[name as keyof T] = function (this: T, ...args: any[]) {
+            try {
+                return func.apply(this, args) || defaultReturn;
```

Comment:
the return value should be create each time. please convert to a function

Reply:
I believe the result is serialized, but I'll recreate it just to be safe.